worlds/checksmate/__init__.py

```diff
@@ -0,0 +1,386 @@
+import math
+import random
+from enum import Enum
+from typing import List, Dict, ClassVar, Callable
+
+from BaseClasses import Tutorial, Region, MultiWorld, Item
+from worlds.AutoWorld import WebWorld, World
+from .Options import cm_options, get_option_value
+from .Items import CMItem, item_table, create_item_with_correct_settings, filler_items, progression_items, useful_items
+from .Locations import CMLocation, location_table
+from .Options import cm_options
+from .Rules import set_rules
+
+
+class CMWeb(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the ChecksMate software on your computer. This guide covers single-player, "
+        "multiworld, and related software.",
+        "English",
+        "checksmate_en.md",
+        "checks-mate/en",
+        ["roty", "rft50"]
+    )]
+
+
+class CMWorld(World):
+    """
+    ChecksMate is a game where you play chess, but all of your pieces were scattered across the multiworld.
+    You win when you checkmate the opposing king!
+    """
+    game: ClassVar[str] = "ChecksMate"
+    option_definitions = cm_options
+    data_version = 0
+    web = CMWeb()
+    required_client_version = (0, 3, 4)
```

Comment:
Is this the actual required version? I think Multiclient.NET reports higher than this.

Reply:
Thisn misunderstand of mine is a primary blocker on my moving the client from prerelease to alpha, as I don't want to emit a versioned release without understanding the relationship of server, client, apworld, and application versions. I'm not sure where to read more, so any advice is appreciated.